File: dialoglibrary/src/main/java/com/sdg/dialoglibrary/test/NormalDialog.java
package com.sdg.dialoglibrary.test;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.view.Gravity;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.sdg.dialoglibrary.R;
import com.sdg.dialoglibrary.utils.SdgUtils;

public class NormalDialog extends Dialog {

    private Context context;

    private String title;
    private String message;
    private String yes = "确定";
    private String no = "取消";

    private float width = 9/10f;

    private Callback callback;
    private TextView tv_title;
    private TextView tv_message;
    private TextView tv_no;
    private TextView tv_yes;

    public NormalDialog(@NonNull Context context) {
        super(context);
        this.context = context;
    }

    public NormalDialog(@NonNull Context context, int themeResId) {
        super(context, themeResId);
        this.context = context;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        View view = View.inflate(context, R.layout.dialog_normal,null);
        setContentView(view);

        Window win = getWindow();
        WindowManager.LayoutParams lp = win.getAttributes();
        lp.gravity = Gravity.CENTER;
        lp.height = WindowManager.LayoutParams.WRAP_CONTENT;
        lp.width = (int) (SdgUtils.getScreenWidthPixels() * width);
        win.setAttributes(lp);

        tv_title = view.findViewById(R.id.tv_title);
        tv_message = view.findViewById(R.id.tv_message);
        tv_no = view.findViewById(R.id.tv_no);
        tv_yes = view.findViewById(R.id.tv_yes);

        initData();

    }

    private void initData(){
        tv_title.setText(title);
        tv_message.setText(message);
        tv_no.setText(no);
        tv_yes.setText(yes);

        tv_no.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                dismiss();
                callback.no();
            }
        });
        tv_yes.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                dismiss();
                callback.yes();
            }
        });
    }

    public NormalDialog setTitle(String title) {
        this.title = title;
        return this;
    }

    public NormalDialog setMessage(String message) {
        this.message = message;
        return this;
    }

    public NormalDialog setYes(String yes) {
        this.yes = yes;
        return this;
    }

    public NormalDialog setNo(String no) {
        this.no = no;
        return this;
    }

    public NormalDialog setWidth(float width) {
        if(width > 1){
            this.width = 9/10f;
            return this;
        }
        this.width = width;
        return this;
    }

    public NormalDialog setCallback(Callback callback) {
        this.callback = callback;
        return this;
    }

    public interface Callback {
        void yes();
        void no();
    }
}

File: dialoglibrary/src/main/java/com/sdg/dialoglibrary/dialog/NormalDialog.kt
package com.sdg.dialoglibrary.dialog

import android.os.Bundle
import android.view.View
import android.view.WindowManager
import android.widget.ImageView
import android.widget.TextView
import com.sdg.dialoglibrary.R
import com.sdg.dialoglibrary.utils.AppUtils

class NormalDialog : BaseDialog() {

    private var confirm: String? = "确定"

    private var cancle: String? = "取消"

    private var title: String? = null

    private var content: String? = null

    private var callback: Callback? = null

    //返回键取消
    private var backCancel: Boolean = false
    //弹框外部取消
    private var canceledOutside: Boolean = false

    override fun getLayoutId(): Int = R.layout.dialog_normal

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        view.findViewById<TextView>(R.id.content).apply {
            text = content
        }

        view.findViewById<ImageView>(R.id.close).apply {
            setOnClickListener {
                dismiss()
            }
        }

        view.findViewById<TextView>(R.id.confirm).apply {
            text = confirm
            setOnClickListener {
                dismiss()
                callback?.onConfirm()
            }
        }

        view.findViewById<TextView>(R.id.cancle).apply {
            text = cancle
            setOnClickListener {
                dismiss()
                callback?.onCancle()
            }
        }
    }

    fun setConfirm(confirm: String): NormalDialog {
        this.confirm = confirm
        return this
    }

    fun setCancle(cancle: String): NormalDialog {
        this.cancle = cancle
        return this
    }

    fun setTitle(title: String): NormalDialog {
        this.title = title
        return this
    }

    fun setContent(content: String): NormalDialog {
        this.content = content
        return this
    }

    fun setBackCancel(backCancel: Boolean): NormalDialog {
        this.backCancel = backCancel
        return this
    }

    fun setOutsideCancel(canceledOutside: Boolean): NormalDialog {
        this.canceledOutside = canceledOutside
        return this
    }

    fun setCallBack(callback: Callback): NormalDialog {
        this.callback = callback
        return this
    }

    override fun windowHeight(): Int {
        return WindowManager.LayoutParams.MATCH_PARENT
    }

    override fun windowWidth(): Int {
        return WindowManager.LayoutParams.MATCH_PARENT
    }

    interface Callback {
        fun onConfirm()
        fun onCancle()
    }

}
